cap/01-categorie.tex

```diff
@@ -1,9 +1,162 @@
+Allo stesso modo, ci sono varie matrici identità, una per ogni $n$. 
+La struttura algebrica che otteniamo è una \emph{categoria}, che definiamo a breve.
+
+Nella definizione di categoria, in generale, è problematico usare la parola ``insieme'', è meglio parlare di \emph{classe}. Informalmente, una classe è come un insieme, ma potenzialmente ``più grande'', cioè può contenere tutti gli insiemi (ma non tutte le classi). 
+Nella definizione che segue, a prima vista si può leggere la parola ``classe'' come ``insieme''. Vedremo la questione più in dettaglio nell'Appendice \ref{fondamenti}. 
+
+\begin{definition}
+ Una \emph{categoria} $\ctC$ consiste dei seguenti dati.
+ \begin{itemize}
+  \item Una classe $\ctC_0$ i cui elementi chiamiamo \emph{oggetti}, di solito indicati con lettere latine maiuscole: $A$, $B$, $X$, $Y$,\dots
+  \item Una classe $\ctC_1$ i cui elementi chiamiamo \emph{morfismi}, di solito indicati con lettere latine minuscole: $f$, $g$, $h$,\dots
+  
+  Queste classi sono fornite delle seguenti strutture. 
+  
+  \item Ad ogni morfismo corrispondono due oggetti $\dom{f}$, $\cod{f}$ chiamati \emph{dominio} e \emph{codominio}. Se $f$ ha dominio $X$ e codominio $Y$, scriviamo $f:X\to Y$, o in forma diagrammatica, $X \xrightarrow{f} Y$.
+  \item Ogni oggetto $X$ ha un particolare morfismo $\id_X:X\to X$ chiamato \emph{identità}.
+  \item Dati tre oggetti $X,Y,Z$ e morfismi $f:X\to Y$ e $g:Y\to Z$ (si noti che $\cod{f}=\dom{g}$) esiste un particolare morfismo $g\circ f:X\to Z$ chiamato \emph{composizione di $f$ e $g$}. Graficamente:
+  $$
+  \begin{tikzcd}
+   X \ar{r}{f} \ar[bend right=20]{rr}[swap]{g\circ f} & Y \ar{r}{g} & Z .
+  \end{tikzcd}
+  $$
+  (Si noti che $\dom{g\circ f}=\dom{f}$ e $\cod{g\circ f}=\cod{g}$.)
+  
+  Queste strutture devono soddisfare le seguenti proprietà.
+  
+  \item \emph{Unitalità}: per ogni morfismo $f:X\to Y$, le composizioni $f\circ\id_X$ e $\id_Y\circ f$ sono uguali ad $f$.
+  \item \emph{Associatività}: dati oggetti $X,Y,Z,A$ e morfismi $f:X\to Y$, $g:Y\to Z$ e $h:Z\to A$, le composizioni $h\circ (g\circ f)$ e $(h\circ g)\circ f$ sono uguali.
+ \end{itemize}
+\end{definition}
+
+Dati due oggetti $X$ e $Y$, indichiamo con $\Hom{\ctC}(X,Y)$ la classe di morfismi da $X$ a $Y$. Altri autori usano altre notazioni, come $\mathrm{Hom}(X,Y)$ o $\mathrm{Hom}_\ctC(X,Y)$.
+
+
+Quando le classi $\ctC_0$ e $\ctC_1$ sono insiemi, chiamamo la categoria $\ctC$ \emph{piccola}. Più in generale, $\ctC$ si dice \emph{localmente piccola} se dati ogni due oggetti $X$ e $Y$, la classe $\Hom{\ctC}(X,Y)$ di morfismi da $X$ a $Y$ è un insieme. 
+
+\begin{examples} 
+ Vediamo alcuni esempi classici di categorie. 
+ \begin{itemize}
```

Comment:
In tutto questo itemize sarei dell'idea di espandere ciascun elemento definendo in maniera molto pedante gli oggetti, i morfismi, le identità, la regola di composizione... Per chi è all'inizio, vedere queste cose fatte esplicitamente aiuta a capire come ci si esprime in linguaggio categoriale.

Reply:
Dove c'è scritto "la categoria degli spazi topologici"?

Reply:
Mi sono spiegato male, non c'è scritto qui, ma quando tu dici "consideriamo la categoria degli spazi topologici" la mia domanda è: con quali morfismi? (questo è ciò che portò Ehresmann alla sua scelta estrema, dato che imbarazzi del genere succedono spesso in topologia, credo...) 

Alcuni teoremi poi sono veri solo per certe scelte di morfismi, non altre: per esempio per CHaus "devi" prendere le mappe proprie per dimostrare la dualità di Gel'fand, e in Ring, se gli omomorfismi mandano 1 in 1 o no è una scelta che cambia la categoria (ti cambia l'oggetto iniziale).